File: frontend/lib/data/models/qna_post.dart
// Q&A 커뮤니티 게시글 모델
class QnaPost {
  final String id;
  final String question;
  final String locale;       // 질문 언어 (en | ja | ko)
  final List<QnaAnswer> answers;
  final DateTime createdAt;

  const QnaPost({
    required this.id,
    required this.question,
    required this.locale,
    required this.answers,
    required this.createdAt,
  });
}

class QnaAnswer {
  final String id;
  final String authorId;
  final String authorNickname;
  final String content;
  final bool isAdopted;
  final DateTime createdAt;

  const QnaAnswer({
    required this.id,
    required this.authorId,
    required this.authorNickname,
    required this.content,
    required this.isAdopted,
    required this.createdAt,
  });
}

File: backend/src/main/java/com/coco/entity/QnaPost.java
package com.coco.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity @Table(name = "qna_posts")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class QnaPost {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @Column(length = 500)
    private String question;

    private String locale;              // 질문 언어

    @OneToMany(mappedBy = "qnaPost", cascade = CascadeType.ALL)
    private List<QnaAnswer> answers;

    @Column(updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    public void prePersist() { this.createdAt = LocalDateTime.now(); }
}
